content/posts/maths/NMPC-Two-Wheel-Bicycle-Model.md

```diff
@@ -0,0 +1,186 @@
+5. **[Python-control](https://python-control.readthedocs.io/en/0.10.1/):** An open-source control systems library for Python.
+6. **[OpEn](https://alphaville.github.io/optimization-engine/):** Optimization Engine in full, is a framework for designing and solving non-linear and non-convex optimisation problems. 
+
+## Notation
+| Symbol                | Expansion              | Comment                     |
+|--------               |-----------             |---------                    |
+|$0_{n \times m}$       | $[0]_{n \times m}$     | An $n \times m$ null matrix.|
+|$1_{n \times m}$       | $[1]_{n \times m}$     | An $n \times m$ matrix with every element equal to 1.|
+|$I_{n}$                | $I_{n \times n}$       | An $n \times n$ identity matrix.|
+|$\R$                   |$(-\infty,\ \infty)$    | Set of real numbers.        |
+|$\R^n$                 |$\\{\mathbf{x}_ {n \times 1}: x_i \in \R \ \forall \ i \in \Z_{[1,\ n]}, n \in \R\\}$| Set of all $n \times 1$ real vectors.|
+|$\R^{n \times m}$      |$\\{\mathbf{A}_ {n \times m}: a_ {i,\ j} \in \R \ \forall \ i \in \Z_{[1,\ n]},\text{ and } j \in \Z_{[1,\ m]}\\}$| Set of all $n \times 1$ real vectors.|
+|$\R_{[i,\ j]}$         |$\\{k: i \le k \le j\ \forall\ i,\ j,\ k \in \R\\}$| Set of real numbers from $i$ to $j$ inclusive.        |
+|$\R_-$ or $\R_{\lt 0}$ |$(-\infty,\ 0)$         | Set of all negative real numbers.|
+|$\R_+$ or $\R_{\ge 0}$ |$[0,\ \infty)$          | Set of all positive real numbers.|
+|$\Z$                   |$(\ldots, -2,\ -1,\ 0,\ 1,\ 2, \ldots)$| Set of all integers.|
+|$\Z_{[i,\ j]}$         |$\\{k: i \le k \le j\ \forall\ i,\ j,\ k \in \Z\\}$| Set of all integers between $i$ and $j$ inclusive.|
+|$\Z_-$ or $\Z_{\lt 0}$ |$(\ldots, -2,\ -1,\ 0)$ | Set of all negative integers.|
+|$\Z_+$ or $\Z_{\ge 0}$ |$[0,\ 1,\ 2, \ldots)$   | Set of all positive integers.|
+|$\mathbb{S}_+^n$       |$\\{\mathbf{A}: \mathbf{x}^\mathsf{T}\mathbf{A}\mathbf{x} \ge 0 \ \forall \ \mathbf{x} \in \R^n, \ \mathbf{A} \in \R^{n \times n},\ \text{and } n \in \R_+ \\}$| Set of all symmetric positive semi-definite matrices.|
+|$\mathbb{S}_{++}^n$    |$\\{\mathbf{A}: \mathbf{x}^\mathsf{T}\mathbf{A}\mathbf{x} \gt 0 \text{ and } \mathbf{x}^\mathsf{T}\mathbf{A}\mathbf{x} \implies \mathbf{x} = 0 \ \forall \ \mathbf{x} \in \R^n, \ \mathbf{A} \in \R^{n \times n},\ \text{and } n \in \R_+ \\}$| Set of all symmetric positive definite matrices.|
+
+
+## Section 1: Initial Formulation
+This section showcases the simulation results of an NMPC controlled vehicle based on bicycle model dynamics used in [1]. The results in this section are limited to the implementation of vehicle dynamics and the fromulation of an NMPC to drive it to a reference position on the global xy-coordinate space. This section does not include autonomous navigation on a track and obstacle avoidance. 
```

Comment:
It would be nice to include a figure to illustrate the various forces etc. Note that taking a figure from the literature is not allowed without permission from the authors, so it's best to make your own figure in PowerPoint.

Reply:
Done.